mdbook-tera-backend/README.md

```diff
@@ -0,0 +1,84 @@
+# [Tera](https://github.com/Keats/tera) backend extension for `mdbook`'s HTML backend
+
+[![Visit crates.io](https://img.shields.io/crates/v/mdbook-i18n-helpers?style=flat-square)](https://crates.io/crates/mdbook-tera-backend)
+[![Build workflow](https://img.shields.io/github/actions/workflow/status/google/mdbook-i18n-helpers/test.yml?style=flat-square)](https://github.com/google/mdbook-i18n-helpers/actions/workflows/test.yml?query=branch%3Amain)
+[![GitHub contributors](https://img.shields.io/github/contributors/google/mdbook-i18n-helpers?style=flat-square)](https://github.com/google/mdbook-i18n-helpers/graphs/contributors)
+[![GitHub stars](https://img.shields.io/github/stars/google/mdbook-i18n-helpers?style=flat-square)](https://github.com/google/mdbook-i18n-helpers/stargazers)
+
+This `mdbook` backend makes it possible to use [tera](https://github.com/Keats/tera)
+templates and expand the capabilities of your books. It works on top of the default HTML
+backend.
+
+## Installation
+
+Run
+
+```shell
+$ cargo install mdbook-tera-backend
+```
+
+## Usage
+
+### Configuring the backend
+
+To enable the backend, simply add `[output.tera-backend]` to your `book.toml`,
+and configure the place where youre templates will live.
+For instance `theme/templates`:
+
+```toml
+...
+
+[output.tera-backend]
+templates_dir = "theme/templates"
+
+...
+```
+
+### Creating templates
+
+Create your template files in the same directory as your book.
+
+```html
+<!-- ./theme/templates/language_list.html -->
+<ul>
+{% for identifier, language_name in get_context(key="output.i18n.languages") %}
+    <li>{{ identifier }}: {{ language_name }}</li>
+{% endfor %}
+</ul>
+```
+
+### Using templates in `index.hbs`
+
+Since the HTML renderer will first render Handlebars templates, we need to tell it to
+ignore tera templates using `{{{{raw}}}}` blocks:
+
+```html
+{{{{raw}}}}
+{% set current_language = ctx | get(key="config") | get(key="book") | get(key="language", default="en") %}
```

Comment:
This can be simplified to just

```suggestion
{% set current_language = ctx.config.book.language %}
```

If you want to handle an unset `language`, the you can do

```
{% set current_language = ctx.config.book.language | default(value="en") %}
```

However, [`mdbook` uses `"en"` by default](https://github.com/rust-lang/mdBook/blob/94e0a44e152d8d7c62620e83e0632160977b1dd5/src/config.rs#L427) (but wraps it in `Some`).

Reply:
You resolved this, but I still see a lot of `ctx | get(key="config") | get(key="book")` above? Why is that?